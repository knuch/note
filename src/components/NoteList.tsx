import React from 'react';
import { Note } from '../data/definitions';
import NoteItem from './NoteItem';

type Props = {
  mode: string;
  currentNote: string | undefined;
  notes: Note[];
}

const NoteList: React.FC<Props> = ({ mode, currentNote, notes }) => {
  return (
    <div className={`note-pane-left ${mode === 'edit' ? 'disabled-zone' : ''}`} >
      {notes.map((note: Note) => (
        <NoteItem
          key={note.id}
          id={note.id}
          title={note.title}
          isCurrent={currentNote ? currentNote === note.id : false}
        />
      ))}
    </div>
  )
}

export default NoteList;
