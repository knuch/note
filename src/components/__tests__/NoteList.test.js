import React from 'react';
import NoteList from '../NoteList';
import { mount } from 'enzyme';

// example of a snapshot-based component test
it('NoteList renders correctly', () => {
  const notes = [
    {
      title: 'My first note',
      id: 'ID-1',
      text: '#Markdown text'
    },
    {
      title: 'My second note',
      id: 'ID-2',
      text: '#Markdown text'
    },
    {
      title: 'My third note',
      id: 'ID-3',
      text: '#Markdown text'
    }
  ];
  const tree = mount(
    <NoteList
      mode='view'
      notes={notes}
      currentNode={undefined}
    />
  );
  expect(tree).toMatchSnapshot();
});
