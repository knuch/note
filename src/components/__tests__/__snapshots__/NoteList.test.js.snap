// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NoteList renders correctly 1`] = `
<NoteList
  mode="view"
  notes={
    Array [
      Object {
        "id": "ID-1",
        "text": "#Markdown text",
        "title": "My first note",
      },
      Object {
        "id": "ID-2",
        "text": "#Markdown text",
        "title": "My second note",
      },
      Object {
        "id": "ID-3",
        "text": "#Markdown text",
        "title": "My third note",
      },
    ]
  }
>
  <div
    className="note-pane-left "
  >
    <NoteListItem
      id="ID-1"
      isCurrent={false}
      key="ID-1"
      title="My first note"
    />
    <NoteListItem
      id="ID-2"
      isCurrent={false}
      key="ID-2"
      title="My second note"
    />
    <NoteListItem
      id="ID-3"
      isCurrent={false}
      key="ID-3"
      title="My third note"
    />
  </div>
</NoteList>
`;
